lib/src/parsers/utilities/type-parser.ts

```diff
@@ -217,6 +220,53 @@ function parseUnionType(typeNode: UnionTypeNode): DataType {
   }
 }
 
+/**
+ * Parse an indexed access type node.
+ *
+ * @param typeNode index access type node
+ */
+function parseIndexedAccessType(typeNode: IndexedAccessTypeNode): TypeNode {
+  const object = typeNode.getObjectTypeNode();
+  const index = typeNode.getIndexTypeNode();
+
+  if (TypeGuards.isIndexedAccessTypeNode(object)) {
+    // FIXME: add support for nested indexed access types
+    throw new Error(
+      "indexed access type error: nested indexed access type unsupported"
+    );
+  }
+
+  if (!TypeGuards.isTypeReferenceNode(object)) {
+    throw new Error("indexed access type error");
```

Comment:
As per, the previous `throw` statement. Can you add a bit more context to the exception(s) being thrown, as in more helpful messages.